v2/pkgs/tools.go

```diff
@@ -166,21 +175,28 @@ func (t *Tools) Install(ctx context.Context, payload *tools.ToolPayload) (*tools
 	var index Index
 	json.Unmarshal(body, &index)
 
-	for _, packager := range index.Packages {
-		if packager.Name != payload.Packager {
-			continue
-		}
-
-		for _, tool := range packager.Tools {
-			if tool.Name == payload.Name &&
-				tool.Version == payload.Version {
-
-				sys := tool.GetFlavourCompatibleWith(runtime.GOOS, runtime.GOARCH)
+	correctTool, correctSystem, found := FindTool(payload.Packager, payload.Name, payload.Version, index)
+	path = filepath.Join(payload.Packager, correctTool.Name, correctTool.Version)
 
-				return t.install(ctx, path, sys.URL, sys.Checksum)
+	key := correctTool.Name + "-" + correctTool.Version
+	// Check if it already exists
+	if t.behaviour == "keep" && pathExists(t.folder) {
```

Comment:
This check is useless I think, since behaviur is set to be `replace` in `v2/http.go`

Reply:
Actually, it can also be set to `keep` if it comes from [here](https://github.com/arduino/arduino-create-agent/pull/953/commits/18e9250110c8185a0d4cff95e0f4b9cd0922b280#diff-8f8dce857327f2cd811b49a13cebb825064a57ed9888bbf2f2dc39ed8494c0b8R48) so this check is necessary.